app/controllers/admin_general_controller.rb

```diff
@@ -74,16 +84,28 @@ def timeline
     # Note that the relevent date for InfoRequestEvents is creation, but
     # for PublicBodyVersions is update thoughout
     connection = InfoRequestEvent.connection
-    timestamps = connection.select_rows("SELECT id,'InfoRequestEvent',
-                                                    created_at AS timestamp
-                                             FROM info_request_events
-                                             WHERE created_at > '#{date_back_to.getutc}'
-                                             UNION
-                                             SELECT id, 'PublicBodyVersion',
-                                                  updated_at AS timestamp
-                                             FROM #{PublicBody.versioned_class.table_name}
-                                             WHERE updated_at > '#{date_back_to.getutc}'
-                                             ORDER by timestamp desc")
+    authority_change_clause = "SELECT id, 'PublicBodyVersion',
+                                    updated_at AS timestamp
+                               FROM #{PublicBody.versioned_class.table_name}
+                               WHERE updated_at > '#{date_back_to.getutc}'"
+
+    info_request_event_clause = "SELECT id,'InfoRequestEvent',
+                                        created_at AS timestamp
+                                 FROM info_request_events
+                                 WHERE created_at > '#{date_back_to.getutc}'"
+
+    if params[:event_type] == 'authority_change'
+      timestamps = connection.select_rows("#{authority_change_clause}
+                                          ORDER by timestamp desc")
+    elsif params[:event_type] == 'info_request_event'
+      timestamps = connection.select_rows("#{info_request_event_clause}
+                                          ORDER by timestamp desc")
+    else
+      timestamps = connection.select_rows("#{info_request_event_clause}
+                                           UNION
+                                           #{authority_change_clause}
+                                           ORDER by timestamp desc")
+    end
```

Comment:
In general, this action is getting really long - we could split out a lot of this new stuff into some private methods?  `get_event_type`, `get_events_title`, `get_date_back_to_utc` (avoiding the need to convert it inside the SQL then) and `get_timestamps` perhaps?